File: march-2026/3-20-cherry-blossoms/cherry-blossoms.py
def cherry_blossoms(temps):
  # for i starting at index 4
  # since we need at least 5 days of data
  for i in range(4, len(temps)):
    # current window of 5 days
    # taking last 4 days + today
    window = temps[i-4:i+1]

    # if the average is at least 15
    # return the day
    if sum(window) / 5 >= 15:
      return i + 1
  
  # else return -1
  return -1

File: march-2026/3-20-cherry-blossoms/cherry_blossoms.js
// Cherry Blossoms 🌸
// shercodes

function cherryBlossoms(temps) {
  // Defines the variable for bloomDay.
  let bloomDay = -1; // -1 when there's no bloom day.

  for (let i = 4; i < temps.length; i++) {
    
    // Adds the current number plus the 4 before it.
    let sum = 0;
    for (let j = i - 4; j <= i; j++) {
      sum += temps[j];
    }

    // Calculates the 5-day average. 
    let avg = sum / 5;

    // Compares if the average is greater or equal to 15.
    if (avg >= 15) {
      bloomDay = i + 1; // i + 1 when there's a bloom day.
      return bloomDay;
    }
  }

  return bloomDay;
};